app/src/main/java/com/example/ubclaunchpad/launchnote/photoBrowser/AllPhotosFragment.kt

```diff
@@ -75,6 +80,45 @@ class AllPhotosFragment : Fragment() {
         }
     }
 
+    fun saveSelection() {
+                Flowable.fromCallable {
+                        LaunchNoteDatabase.getDatabase(activity)?.let {
+                                picNotesSelected.forEach { pn ->
+                                       Log.i("SAV", "saving: " + pn)
+                                        if (ContextCompat.checkSelfPermission(getActivity(),
+                                                        android.Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED) {
+
+                                                var bitmap: Bitmap;
+                                                 bitmap = Glide.with(activity)
+                                                         .asBitmap()
+                                                         .load(pn.imageUri)
+                                                         .submit()
+                                                         .get();
+
+                                                Log.i("SAV", "BITMAP" + bitmap)
+                                                MediaStore.Images.Media.insertImage(activity.getContentResolver(), bitmap , "yourTitle", "yourDescription");
```

Comment:
So many indents, would you mind breaking it down to smaller functions. I think you can define them inside saveSelection